src/Randomizer.SMZ3.Tracking/HistoryService.cs

```diff
@@ -19,38 +20,29 @@ namespace Randomizer.SMZ3.Tracking.VoiceCommands
     /// </summary>
     public class HistoryService : IHistoryService
     {
-        private ILogger<HistoryService> _logger;
+        private readonly ILogger<HistoryService> _logger;
+        private readonly IWorldAccessor _world;
+        private ICollection<TrackerHistoryEvent> _historyEvents => _world.World.State.History;
         private Tracker? _tracker;
-        private List<TrackerHistoryEvent>? _events;
 
         /// <summary>
         /// Constructor
         /// </summary>
+        /// <param name="world"></param>
         /// <param name="logger"></param>
-        public HistoryService(ILogger<HistoryService> logger)
+        public HistoryService(IWorldAccessor world, ILogger<HistoryService> logger)
         {
+            _world = world;
             _logger = logger;
         }
 
         /// <summary>
-        /// Starts a new history list for a tracker instance
+        /// Sets the tracker to be used for getting the elapsed time
         /// </summary>
-        /// <param name="tracker">The tracker instance</param>
-        public void StartHistory(Tracker tracker)
+        /// <param name="tracker"></param>
+        public void SetTracker(Tracker tracker)
```

Comment:
Can't we inject Tracker now?

Reply:
Unfortunately, this service is actually injected by Tracker, so I think I ran into issues with that because of it being cyclic injections. That being said, the only thing the HistoryService actually uses from Tracker is to get the time stamp. I can probably just make a TrackingTimerService for managing starting/pausing the timer and getting the current elapsed time. That would be able to be injected by both Tracker and this.